src/store/video-slice.js

```diff
@@ -0,0 +1,101 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+import { baseFetch } from '../api/baseFetch'
+import { getLessons } from './materials-slice'
+
+export const addVideo = createAsyncThunk(
+   'videos/addVideo',
+   async ({ video, lessonId }, { rejectWithValue, dispatch }) => {
+      try {
+         const response = await baseFetch({
+            path: `api/videos/${lessonId}`,
+            method: 'POST',
+            body: video,
+         })
+         dispatch(getLessons())
+         return response
+      } catch (error) {
+         return rejectWithValue(error.message)
+      }
+   }
+)
+export const deleteVideo = createAsyncThunk(
+   'videos/deleteVideo',
+   async (id, { rejectWithValue, dispatch }) => {
+      try {
+         const response = await baseFetch({
+            path: `api/videos/${id}`,
+            method: 'DELETE',
+         })
+         dispatch(getLessons())
+         return response
+      } catch (error) {
+         return rejectWithValue(error.message)
+      }
+   }
+)
+export const getSingleVideo = createAsyncThunk(
+   'videos/getSingleVideo',
+   async (id, { rejectWithValue, dispatch }) => {
+      try {
+         const response = await baseFetch({
+            path: `api/videos/${id}`,
+            method: 'GET',
+         })
+         dispatch(videoActions.getSingleVideo(response))
```

Comment:
we already fetched lesson than action.setVideo or setSomething not getSommething